KAFKA-14398: Update EndToEndAuthorizationTest to test both ZK and KRAFT quorum servers

pprovenzano opened the pull request:
All EndToEndAuthorizationTests pass in both modes, but
SCRAM and Delegation are not implemented for KRAFT yet 
so they emit a message to stderr and pass the test.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

pprovenzano committed:
Update EndToEndAuthorizationTest to test both ZK and KRAFT quorum servers

SCRAM and Delegation are not implemented for KRAFT yet so they emit
a message to stderr and pass the test.

omkreddy commented on core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala:
```diff
@@ -36,15 +37,33 @@ import org.apache.kafka.common.errors.{GroupAuthorizationException, TopicAuthori
 import scala.jdk.CollectionConverters._
 
+/**
+  * All broker to controller communications is authenticated as PLAINTEXT in these tests.
```
For broker -> controller communication, we forward the authenticated principal (`User:server`) on broker to controller nodes using forward request mechanism. 

we can include "User:ANONYMOUS" to `super.users` config to allow PLAINTEXT communication between  controller nodes communication.

pprovenzano replied:
Okay, I fixed this. Thanks for the tip.

omkreddy reviewed (commented):
looks we still need to convert `testNoDescribeProduceOrConsumeWithoutTopicDescribeAcl` `testNoProduceWithDescribeAcl`  tests.

Do we also need convert all classes tests.

pprovenzano committed:
Remove TestControllerPrincipalBuilder and instead set superuser to
Anonymous on the contollers only.

Update a couple tests that now need multiple test params to test kraft

pprovenzano committed:
Update a couple more tests to work with KRAFT Quorum servers.

pprovenzano commented:
I have validated that the failing tests work by manually running them.

omkreddy reviewed (approved):
@pprovenzano Thanks for the PR. LGTM
